backend/remote/backend_test.go

```diff
@@ -64,6 +64,18 @@ func TestRemote_config(t *testing.T) {
 			}),
 			confErr: "Failed to request discovery document",
 		},
+		"without_a_token": {
+			config: cty.ObjectVal(map[string]cty.Value{
+				"hostname":     cty.StringVal("localhost"),
```

Comment:
One of the two supported hostnames in the test disco services, along with `app.terraform.io`: https://github.com/hashicorp/terraform/blob/e57685d8fc1a065ab5045044056550b50e2cd9e4/backend/remote/testing.go#L276-L277

There's no token for `localhost` by default, which triggers the error we want: https://github.com/hashicorp/terraform/blob/e57685d8fc1a065ab5045044056550b50e2cd9e4/backend/remote/testing.go#L37-L39

This happily also allows us to verify the case where the login command needs to specify the hostname.

Reply:
This is great info and might make a good comment right above this test case:
`// no token is returned for localhost`  (or something to that effect)

Reply:
Thanks, done! This is a good lesson for me in general—if it's worth a PR self-comment, it's probably also worth a code comment.